ONSdigital/ras-party
# Modify party search

# Motivation and Context
The search for business information is very slow, especially when few or no search terms are included in the search. 

# What has changed
- name and trading as columns added to business_attributes table
- scripts added to add new columns and to put example data into the database
- search_business function modified to be a bit smarter about searching for a full ru_ref while making every other type of search use the column data instead of faster.

# How to test?
- Delete docker image for postgres
- rebuild docker containers, look at the partysvc -> business_attributes table.  The name and trading_as columns should be added (these are generated from the model when the party schema is missing)

- Next, seed some data with the acceptance tests setup.
- sign into response-operations-ui and search for a reporting unit.  Everything should still work.

- Finally (to test the speed), go to the scripts directory, and `pipenv run python insert_into_party.py`.  By default this will add 20k businesses and 100k rows in to the business attributes table but you can change the numbers.  Do this a few times then do an empty search, a search with a few characters and an ru_ref search.  These should all be reasonably fast

# Links
https://trello.com/c/WerjgQnt
